packages/router/server/src/provide-server-context.ts

```diff
@@ -0,0 +1,56 @@
+import { ɵSERVER_CONTEXT as SERVER_CONTEXT } from '@angular/platform-server';
+
+import {
+  BASE_URL,
+  REQUEST,
+  RESPONSE,
+  ServerRequest,
+  ServerResponse,
+} from '@analogjs/router/tokens';
+
+export function providerServerContext({
+  req,
+  res,
+}: {
+  req: ServerRequest;
+  res: ServerResponse;
+}) {
+  const baseUrl = getBaseUrl(req);
+
+  return [
+    { provide: SERVER_CONTEXT, useValue: 'ssr-analog' },
+    { provide: REQUEST, useValue: req },
+    { provide: RESPONSE, useValue: res },
+    { provide: BASE_URL, useValue: baseUrl },
+  ];
+}
+
+export function getBaseUrl(req: ServerRequest) {
+  const protocol = getRequestProtocol(req);
+  const { originalUrl, headers } = req;
+  const parsedUrl = new URL(
+    '',
+    `${protocol}://${headers.host}${
+      originalUrl.endsWith('/')
+        ? originalUrl.substring(0, originalUrl.length - 1)
+        : originalUrl
+    }`
+  );
+  const baseUrl = parsedUrl.origin;
+
+  return baseUrl;
+}
+
+export function getRequestProtocol(
```

Comment:
fyi, I took this function from here 😄 https://github.com/unjs/h3/blob/bd7831bde071d368cf3f120128cc1823bfb4c3c8/src/utils/request.ts#L238

But I implemented it in the code because Initially, I hadn't access to the `H3Event`

Reply:
In development, we won't have access to the `H3Event`, but the `req` and `res` are the same with the Vite server in development, and Nitro in the production server.